mmdet/models/dense_heads/maskformer_head.py

```diff
@@ -0,0 +1,716 @@
+            assert isinstance(class_weight, float), 'Expected ' \
+                'class_weight to have type float. Found ' \
+                f'{type(class_weight)}.'
+            # NOTE following the official MaskFormerHead repo, bg_cls_weight
+            # means relative classification weight of the VOID class.
+            bg_cls_weight = loss_cls.get('bg_cls_weight', class_weight)
+            assert isinstance(bg_cls_weight, float), 'Expected ' \
+                'bg_cls_weight to have type float. Found ' \
+                f'{type(bg_cls_weight)}.'
+            class_weight = torch.ones(self.num_classes + 1) * class_weight
+            # set VOID class as the last indice
+            class_weight[self.num_classes] = bg_cls_weight
+            loss_cls.update({'class_weight': class_weight})
+            if 'bg_cls_weight' in loss_cls:
+                loss_cls.pop('bg_cls_weight')
+            self.bg_cls_weight = bg_cls_weight
+        self.loss_cls = build_loss(loss_cls)
+        self.loss_mask = build_loss(loss_mask)
+        self.loss_dice = build_loss(loss_dice)
+
+    def init_weights(self):
+        if isinstance(self.decoder_input_proj, Conv2d):
+            caffe2_xavier_init(self.decoder_input_proj, bias=0)
+
+        self.pixel_decoder.init_weights()
+
+        for p in self.transformer_decoder.parameters():
+            if p.dim() > 1:
+                nn.init.xavier_uniform_(p)
+
+    def preprocess_gt(self, gt_labels_list, gt_masks_list, gt_semantic_segs):
+        """Preprocess the ground truth for all images.
+
+        Args:
+            gt_labels_list (list[Tensor]): Each is ground truth
+                labels of each bbox, with shape (num_gts, ).
+            gt_masks_list (list[BitmapMasks]): Each is ground truth
+                masks of each instances of a image, shape
+                (num_gts, h, w).
+            gt_semantic_seg (Tensor): Ground truth of semantic
+                segmentation with the shape (batch_size, n, h, w).
+                [0, num_thing_class - 1] means things,
+                [num_thing_class, num_class-1] means stuff,
+                255 means VOID.
+            target_shape (tuple[int]): Shape of output mask_preds.
+                Resize the masks to shape of mask_preds.
+
+        Returns:
+            tuple: a tuple containing the following targets.
+
+                - labels (list[Tensor]): Ground truth class indices for all
+                    images. Each with shape (n, ), n is the sum of number
+                    of stuff type and number of instance in a image.
+                - masks (list[Tensor]): Ground truth mask for each image, each
+                    with shape (n, h, w).
+        """
+        targets = multi_apply(self._preprocess_gt_single, gt_labels_list,
+                              gt_masks_list, gt_semantic_segs)
+        labels, masks = targets
+        return labels, masks
+
+    def _preprocess_gt_single(self, gt_labels, gt_masks, gt_semantic_seg):
```

Comment:
The preprocess of GT semantic seg may be done in an independent function, which could be reused by many models like maskformer/mask2former/K-Net etc.